python-frontend/src/main/java/org/sonar/python/quickfix/PythonTextEdit.java

```diff
@@ -43,12 +43,22 @@ public static PythonTextEdit insertBefore(Tree tree, String textToInsert) {
     return new PythonTextEdit(textToInsert, token.line(), token.column(), token.line(), token.column());
   }
 
+
   public static PythonTextEdit insertAfter(Tree tree, String textToInsert) {
     Token token = tree.firstToken();
     int lengthToken = token.value().length();
     return new PythonTextEdit(textToInsert, token.line(), token.column() + lengthToken, token.line(), token.column() + lengthToken);
   }
 
+  public static PythonTextEdit replaceAt(Tree toReplace, String replacementText) {
+    Token token = toReplace.firstToken();
+    return new PythonTextEdit(replacementText, token.line(), token.column(), token.line(), token.column() +token.value().length());
```

Comment:
The calculation of the end offset will not work for expressions like function calls.
The value length of `foo()` is 3, not 5 so you will keep the parentheses. I would use the `lastToken` of `toReplace` to get the right offset.